### PR TITLE
aws-lambda | add IoT Custom Authorizer

### PR DESCRIPTION
**Pull request by rednes**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <<[url here](https://docs.aws.amazon.com/iot/latest/developerguide/config-custom-auth.html#custom-auth-lambda)>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/aws-lambda/test/iot-authorizer-tests.ts
Comment (simonbuchan) [inline types/aws-lambda/test/iot-authorizer-tests.ts:5]: Could you check the formatting here? The prettier check on CI isn't complaining, which is surprising, but it would be good to avoid dinging a future PR.

File: types/aws-lambda/trigger/iot-authorizer.d.ts
Comment (simonbuchan) [inline types/aws-lambda/trigger/iot-authorizer.d.ts:2]: We should probably pull out the certificate/policy stuff to somewhere; they get used all over the place. But not in this PR!

File: types/aws-lambda/trigger/iot-authorizer.d.ts
Comment (simonbuchan) [inline types/aws-lambda/trigger/iot-authorizer.d.ts:6]: Technically there should also be a Callback type alias here as well for consistency, but honestly they should all be marked deprecated at this point, even if the callback parameter itself isn't.

Anyone else have an opinion here?

File: types/aws-lambda/trigger/iot-authorizer.d.ts
Comment (simonbuchan) [inline types/aws-lambda/trigger/iot-authorizer.d.ts:20]: The comment here seems to imply this should be optional? But from what I've done with mqtt I don't think the client id is optional anyway, so I'm rather confused.

**Review by simonbuchan — APPROVED** [review]
Thanks, looks good! Nothing blocking, so I'm approving, but maybe give other reviewers a chance to take a look before merging.

**Review by simonbuchan — APPROVED** [review]

**Comment by rbuckton** [comment]
@simonbuchan: Should I continue to hold off on approving/merging this PR, per your comment here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66818#pullrequestreview-1641690377

**Comment by simonbuchan** [comment]
A week is more than fine!

**Review by rbuckton — APPROVED** [review]

**Comment by rednes** [comment]
Ready to merge

**Pull request merged**
